packages/web/src/components/create-playlist/PlaylistForm.tsx

```diff
@@ -0,0 +1,258 @@
+import UploadArtwork from 'components/upload/UploadArtwork'
+import { useCollectionCoverArt } from 'hooks/useCollectionCoverArt'
+import * as schemas from 'schemas'
+import { resizeImage } from 'utils/imageProcessingUtil'
+
+import styles from './PlaylistForm.module.css'
+
+const messages = {
+  editPlaylistButtonText: 'Save Changes',
+  cancelButtonText: 'Cancel',
+  deletePlaylistButtonText: 'Delete Playlist',
+  deleteAlbumButtonText: 'Delete Album',
+  createPlaylistButtonText: 'Create Playlist'
+}
+
+type PlaylistFormFields = Partial<Collection> & {
+  artwork: {
+    file: Blob
+    url: string
+    source: 'unsplash' | 'original'
+    error?: string
+  }
+  is_current: boolean
+  mood: Nullable<string>
+  created_at: Nullable<string>
+  tags: Nullable<string>
+  genre: Nullable<string>
+  isAlbum: boolean
+} & DeepNullable<
+    Pick<
+      CollectionMetadata,
+      | 'is_private'
+      | 'updated_at'
+      | 'cover_art'
+      | 'cover_art_sizes'
+      | 'playlist_name'
+      | 'playlist_owner_id'
+      | 'save_count'
+      | 'upc'
+      | 'description'
+    >
+  >
+
+type PlaylistFormProps = {
+  metadata?: Nullable<Collection>
+  isAlbum?: boolean
+  onOpenArtworkPopup?: () => void
+  onCloseArtworkPopup?: () => void
+  isEditMode?: boolean
+  /** Only applies to edit mode */
+  onDelete?: () => void
+  /** Only applies to edit mode */
+  onCancel?: () => void
+  onSave: (formFields: PlaylistFormFields) => void
+}
+
+const EditPlaylistActions = ({
+  isAlbum,
+  onDelete,
+  onCancel,
+  onSave
+}: {
+  isAlbum: boolean
+  onDelete?: () => void
+  onCancel?: () => void
+  onSave: () => void
+}) => {
+  return (
+    <div className={styles.editPlaylistActionsContainer}>
+      <div>
+        <Button
+          text={
+            isAlbum
+              ? messages.deleteAlbumButtonText
+              : messages.deletePlaylistButtonText
+          }
+          type={ButtonType.SECONDARY}
+          onClick={onDelete}
+          className={styles.deleteButton}
+          textClassName={styles.deleteButtonText}
+        />
+      </div>
+      <div>
+        <Button
+          text={messages.cancelButtonText}
+          type={ButtonType.SECONDARY}
+          className={styles.cancelButton}
+          textClassName={styles.cancelButtonText}
+          onClick={onCancel}
+        />
+        <Button
+          className={styles.saveChangesButton}
+          text={messages.editPlaylistButtonText}
+          type={ButtonType.SECONDARY}
+          onClick={debounce(onSave, 500, { leading: true })}
+        />
+      </div>
+    </div>
+  )
+}
+
+const CreatePlaylistActions = ({ onSave }: { onSave: () => void }) => {
+  return (
+    <div className={styles.createPlaylistActionsContainer}>
+      <Button
+        rightIcon={<IconCheck />}
+        text={messages.createPlaylistButtonText}
+        type={ButtonType.PRIMARY}
+        onClick={debounce(onSave, 500, { leading: true })}
+      ></Button>
```

Comment:
not sure if prettier is working in monorepo cause i think that should be self-closing

Reply:
I'll add that to lint rules in a separate pr